src/serve_rendered.js

```diff
@@ -347,11 +356,19 @@ const precisePx = (ll, zoom) => {
  * @param {object} marker Marker object parsed by extractMarkersFromQuery.
  * @param {number} z Map zoom level.
  */
-const drawMarker = (ctx, marker, z) => {
-  return new Promise((resolve) => {
+const drawMarker = (ctx, marker, z, scale) => {
+  return new Promise(async (resolve) => {
     const img = new Image();
     const pixelCoords = precisePx(marker.location, z);
 
+    let imgSrc = marker.icon;
+    if (marker.icon === 'default') {
+      // use default built-in marker (svg -> dataURL)
+      const markerDataURL = await generateMarker(scale, marker.color);
+      marker.scale = marker.scale ? marker.scale * (1 / scale) : 1 / scale;
```

Comment:
We do this to handle the markers at different resolutions while still allowing us to scale each individual marker